src/components/collapsible/Collapsible.js

```diff
@@ -72,8 +74,16 @@ class Collapsible extends React.PureComponent<Props, State> {
         );
     };
 
+    handleMouseEnter = () => {
```

Comment:
I think you should try to do this with just CSS, if possible. You can always render the button, with visibility:hidden when its collapsed. And when open, visible. Likewise when hovering.

Reply:
You may also want to consider the focus use case when it is collapsed, and the user tabs into the header with keyboard navigation

Reply:
In this component, Button is passed via headerActionItems pro and isOpen prop controls the visibility of this button. I don't think with just CSS the hover effect can be controlled from where ever its consumed.

Reply:
```jsx
{(isOpen && headerActionItems) || (!isOpen && isHovered && headerActionItems)}
```

can be changed to

```jsx
<span className="this_class_can_be_manupilated">{headerActionItems}</span>
```

Reply:
`{!!headerActionItems && <span className="this_class_can_be_manupilated">{headerActionItems}</span>}` this should help to accommodate the UX requirement with just the CSS. With this change the Button gets displayed all the time (collapsed or not). Shield is using this component I guess. Once this is out they may have to put some CSS to maintain the existing behavior. I am trying to test this out. will be updating the PR soon